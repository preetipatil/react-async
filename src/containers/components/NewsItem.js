import React from 'react';

const NewsItem = ({ article }) => (
  <article>
    <div>
      <h3>
        <a href={article.url}>{article.title}</a>
      </h3>
      <img src={article.urlToImage} alt="" style={{ width: 300, height: 200 }} />
      <p>{article.description}</p>
    </div>
  </article>
);

export default NewsItem;
