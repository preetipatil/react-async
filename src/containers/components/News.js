import React from 'react';
import NewsItem from './NewsItem';
import PropTypes from 'prop-types';

export function NoNews() {
  return <h3>No News Available...</h3>;
}

let News = ({ news }) => {
  if (news) {
    return news.map((article, index) => (
      <div key={`${index}`}>
        <NewsItem article={news[index]} />
      </div>
    ));
  } else {
    return <NoNews />;
  }
};

News.propTypes = {
  news: PropTypes.array,
};
export default News;
