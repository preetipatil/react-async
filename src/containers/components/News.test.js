import React from 'react';
import { shallow } from 'enzyme';
import News from './News';
import NewsItem from './NewsItem';
import { NoNews } from './News';

describe('News Component', () => {
  it('should render News Items correctly', () => {
    const props = {
      news: [
        {
          author: 'BBC News',
          title: 'Japan braced for biggest typhoon in decades',
          description: 'Typhoon Hagibis could be the worst to hit Japan since 1958 when more than 1,200 people died.',
          url: 'http://www.bbc.co.uk/news/world-asia-50020108',
          urlToImage: 'https://ichef.bbci.co.uk/news/1024/branded_news/78D2/production/_109203903_057241635.jpg',
        },
      ],
    };
    const treeWrapper = shallow(<News {...props} />);
    expect(treeWrapper).toMatchSnapshot();
    expect(treeWrapper.containsAnyMatchingElements([<NewsItem />])).toEqual(true);
  });

  it('should show NoNews when no news provided', () => {
    const props = {};
    const treeWrapper = shallow(<News {...props} />);
    //expect(treeWrapper).toMatchSnapshot();
    expect(treeWrapper.containsAnyMatchingElements([<NoNews />])).toEqual(true);
  });
});
