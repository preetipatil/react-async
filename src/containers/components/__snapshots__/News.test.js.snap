// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Component should render News Items correctly 1`] = `
<div
  key="0"
>
  <NewsItem
    article={
      Object {
        "author": "BBC News",
        "description": "Typhoon Hagibis could be the worst to hit Japan since 1958 when more than 1,200 people died.",
        "title": "Japan braced for biggest typhoon in decades",
        "url": "http://www.bbc.co.uk/news/world-asia-50020108",
        "urlToImage": "https://ichef.bbci.co.uk/news/1024/branded_news/78D2/production/_109203903_057241635.jpg",
      }
    }
  />
</div>
`;
